### PR TITLE
Added select-all checkbox in splashpages form

### PR DESCRIPTION
**Pull request by snpd25**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [ ] The tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [ ] I have added necessary documentation (if appropriate).

The "select all" checkbox is added for the quick
selection or deselection of the checkboxes in
admin/splashpages#new.

closes issue #2022

Screenshot from 2019-03-09 09-32-48

-

### CODE REVIEW

**Comment by snpd25** [comment]
@Ana06 , please review it.

File: app/views/admin/splashpages/_form.html.haml [hunk 1]
```diff
@@ -8,6 +9,12 @@
     .col-md-12
       = f.inputs name: 'Components' do
         %ul.fa-ul
+          %li
+            .boolean.input.optional.form-group.checkbox
```
Comment (hennevogel) [inline app/views/admin/splashpages/_form.html.haml]: Please not make this another checkbox in the list but  a button.

**Comment by snpd25** [comment]
Now it looks like:
image

File: app/views/admin/splashpages/_form.html.haml [hunk 2]
```diff
@@ -8,6 +8,8 @@
     .col-md-12
       = f.inputs name: 'Components' do
         %ul.fa-ul
+          %li
+            <th><button type="button" id="selectAll" >Select All</button></th>
```
Comment (hennevogel) [inline app/views/admin/splashpages/_form.html.haml]: please use bootstrap.

**Comment by snpd25** [comment]
@hennevogel , please review it.

File: app/views/admin/splashpages/_form.html.haml [hunk 3]
```diff
@@ -8,6 +8,10 @@
     .col-md-12
       = f.inputs name: 'Components' do
         %ul.fa-ul
+          %li
+            %th
+              %button.btn#selectAll{type: "button"}
```
Comment (hennevogel) [inline app/views/admin/splashpages/_form.html.haml]: Just the `btn` class is not enough. Please read the [bootstrap documentation](https://getbootstrap.com/docs/3.4/css/#buttons).

Comment (hennevogel) [reply app/views/admin/splashpages/_form.html.haml]: I would suggest `btn-default` as styling class. What do you think?

Comment (snpd25) [reply app/views/admin/splashpages/_form.html.haml]: I had tried that, it just looked fine to me without adding the styling class as I was a bit confused about how to style it, `btn-default` is also a good option. Thanks.

Comment (Ana06) [reply app/views/admin/splashpages/_form.html.haml]: @snpd25 apart from looking good, it has to look as the rest of the app :wink:

File: app/views/admin/splashpages/_form.html.haml [hunk 4]
```diff
@@ -45,3 +49,13 @@
     .col-md-12
       %p.text-right
         = f.submit 'Save Changes', class: 'btn btn-primary'
+
+:javascript
+  $(document).ready(function(){
+    var clicked = true;
```
Comment (Ana06) [inline app/views/admin/splashpages/_form.html.haml]: why do  you need this variable? you can use `true` directly

**Comment by differentreality** [comment]
@snpd25 there are also some linters failing (you can see that at the end of his PR), please fix them too.

**Comment by snpd25** [comment]
Here is the screenshot for final changes. :)
image

**Pull request merged**